py/server/deephaven/experimental/partitioned_table_service.py

```diff
@@ -0,0 +1,326 @@
+        """ Provides a callback for the backend service to pass the changed size of the partition with the given
+        table key and partition location key. The callback should be called with the size of the partition in number of
+        rows.
+
+        This is called for tables created when ：meth:`PythonTableDataService.make_table` is called with live=True
+
+        Note that the callback must be called with the initial size of the partition after this method has returned and
+        must not be called before this method has returned.
+
+        The return value is a function that can be called to unsubscribe from the partition size changes.
+
+        Args:
+            table_key (TableKey): the table key
+            table_location_key (PartitionedTableLocationKey): the partition location key
+            callback (Callable[[int], None]): the callback function
+
+        Returns:
+            Callable[[], None]: a function that can be called to unsubscribe from the partition size changes
+        """
+        pass
+
+    @abstractmethod
+    def column_values(self, table_key: TableKey, table_location_key: PartitionedTableLocationKey, col: str, offset: int,
+                      min_rows: int, max_rows: int) -> pa.Table:
+        """ Returns the values for the column with the given name for the partition with the given table key and
+        partition location key. The returned pa.Table should have a single column with values of the specified range
+        requirement for the given column.
+
+        Args:
+            table_key (TableKey): the table key
+            table_location_key (PartitionedTableLocationKey): the partition location key
+            col (str): the column name
+            offset (int): the starting row index
+            min_rows (int): the minimum number of rows to return
+            max_rows (int): the maximum number of rows to return
+
+        Returns:
+            pa.Table: a pa.Table that contains the values for the column
+        """
+        pass
+
+
+class PythonTableDataService(JObjectWrapper):
+    """ A Python wrapper for the Java PythonTableDataService class. It also serves as an adapter between the Java backend
+    interface and the Python backend interface.
+    """
+    j_object_type = _JPythonTableDataService
+    _backend: PartitionedTableServiceBackend
+
+    def __init__(self, backend: PartitionedTableServiceBackend):
+        """ Creates a new PythonTableDataService with the given user-implemented backend service.
+
+        Args:
+            backend (PartitionedTableServiceBackend): the user-implemented backend service implementation
+        """
+        self._backend = backend
+        self._j_tbl_service = _JPythonTableDataService.create(self)
+
+    @property
+    def j_object(self):
+        return self._j_tbl_service
+
+    def make_table(self, table_key: TableKey, *, live: bool) -> Table:
```

Comment:
if `live` doesn't have a default and is always required, why put it after `*`?

Reply:
Because it is more clear what a boolean argument means.